### ISSUE TITLE
The table is not aligned for some _deco configuration

### ISSUE THREAD

**Issue by paul-shuvo** [foutaise/texttable#76]
I ran the following to check all the visual configuration of the table:

```python
for i in range(16):
    table.set_deco(i)
    print(table.draw())
    print(table._deco)
```
The output was the following:
```
0
  Name     Age   Nickname
Mr         32    Xav'    
Xavier                   
Huon                     
Mr         1     Baby    
Baptiste                 
Clement                  
Mme        28    Lou     
Louise                   
Bourgeau         Loue    
1
+---------------------------+
|   Name     Age   Nickname |
| Mr         32    Xav'     |
| Xavier                    |
| Huon                      |
| Mr         1     Baby     |
| Baptiste                  |
| Clement                   |
| Mme        28    Lou      |
| Louise                    |
| Bourgeau         Loue     |
+---------------------------+
2
  Name     Age   Nickname
=========================
Mr         32    Xav'    
Xavier                   
Huon                     
Mr         1     Baby    
Baptiste                 
Clement                  
Mme        28    Lou     
Louise                   
Bourgeau         Loue    
3
+---------------------------+
|   Name     Age   Nickname |
+===========================+
| Mr         32    Xav'     |
| Xavier                    |
| Huon                      |
| Mr         1     Baby     |
| Baptiste                  |
| Clement                   |
| Mme        28    Lou      |
| Louise                    |
| Bourgeau         Loue     |
+---------------------------+
4
  Name     Age   Nickname
Mr         32    Xav'    
Xavier                   
Huon                     
+---------------------------+
Mr         1     Baby    
Baptiste                 
Clement                  
+---------------------------+
Mme        28    Lou     
Louise                   
Bourgeau         Loue    
5
+---------------------------+
|   Name     Age   Nickname |
| Mr         32    Xav'     |
| Xavier                    |
| Huon                      |
+---------------------------+
| Mr         1     Baby     |
| Baptiste                  |
| Clement                   |
+---------------------------+
| Mme        28    Lou      |
| Louise                    |
| Bourgeau         Loue     |
+---------------------------+
6
  Name     Age   Nickname
=========================
Mr         32    Xav'    
Xavier                   
Huon                     
+---------------------------+
Mr         1     Baby    
Baptiste                 
Clement                  
+---------------------------+
Mme        28    Lou     
Louise                   
Bourgeau         Loue    
7
+---------------------------+
|   Name     Age   Nickname |
+===========================+
| Mr         32    Xav'     |
| Xavier                    |
| Huon                      |
+---------------------------+
| Mr         1     Baby     |
| Baptiste                  |
| Clement                   |
+---------------------------+
| Mme        28    Lou      |
| Louise                    |
| Bourgeau         Loue     |
+---------------------------+
8
  Name   | Age | Nickname
Mr       | 32  | Xav'    
Xavier   |     |         
Huon     |     |         
Mr       | 1   | Baby    
Baptiste |     |         
Clement  |     |         
Mme      | 28  | Lou     
Louise   |     |         
Bourgeau |     | Loue    
9
+---------------------------+
|   Name   | Age | Nickname |
| Mr       | 32  | Xav'     |
| Xavier   |     |          |
| Huon     |     |          |
| Mr       | 1   | Baby     |
| Baptiste |     |          |
| Clement  |     |          |
| Mme      | 28  | Lou      |
| Louise   |     |          |
| Bourgeau |     | Loue     |
+---------------------------+
10
  Name   | Age | Nickname
=========+=====+=========
Mr       | 32  | Xav'    
Xavier   |     |         
Huon     |     |         
Mr       | 1   | Baby    
Baptiste |     |         
Clement  |     |         
Mme      | 28  | Lou     
Louise   |     |         
Bourgeau |     | Loue    
11
+---------------------------+
|   Name   | Age | Nickname |
+==========+=====+==========+
| Mr       | 32  | Xav'     |
| Xavier   |     |          |
| Huon     |     |          |
| Mr       | 1   | Baby     |
| Baptiste |     |          |
| Clement  |     |          |
| Mme      | 28  | Lou      |
| Louise   |     |          |
| Bourgeau |     | Loue     |
+---------------------------+
12
  Name   | Age | Nickname
Mr       | 32  | Xav'    
Xavier   |     |         
Huon     |     |         
+---------------------------+
Mr       | 1   | Baby    
Baptiste |     |         
Clement  |     |         
+---------------------------+
Mme      | 28  | Lou     
Louise   |     |         
Bourgeau |     | Loue    
13
+---------------------------+
|   Name   | Age | Nickname |
| Mr       | 32  | Xav'     |
| Xavier   |     |          |
| Huon     |     |          |
+---------------------------+
| Mr       | 1   | Baby     |
| Baptiste |     |          |
| Clement  |     |          |
+---------------------------+
| Mme      | 28  | Lou      |
| Louise   |     |          |
| Bourgeau |     | Loue     |
+---------------------------+
14
  Name   | Age | Nickname
=========+=====+=========
Mr       | 32  | Xav'    
Xavier   |     |         
Huon     |     |         
+---------------------------+
Mr       | 1   | Baby    
Baptiste |     |         
Clement  |     |         
+---------------------------+
Mme      | 28  | Lou     
Louise   |     |         
Bourgeau |     | Loue    
15
+---------------------------+
|   Name   | Age | Nickname |
+==========+=====+==========+
| Mr       | 32  | Xav'     |
| Xavier   |     |          |
| Huon     |     |          |
+---------------------------+
| Mr       | 1   | Baby     |
| Baptiste |     |          |
| Clement  |     |          |
+---------------------------+
| Mme      | 28  | Lou      |
| Louise   |     |          |
| Bourgeau |     | Loue     |
+---------------------------+
```
For 4 and 14, it doesn't align. Usually, the other configurations should suffice, but I just came across the issue when using 14 and thought someone may be able to fix it.

**Comment by foutaise** [comment]
Hi @paul-shuvo

Just made a test and I can't reproduce your issue.
I close this issue for now, feel free to reopen with
more details if needed.

Best regards.

```
$ cat test.py
import texttable

print("texttable version: " + texttable.__version__)
print()

table = texttable.Texttable()
table.set_cols_align(["l", "r", "c"])
table.set_cols_valign(["t", "m", "b"])
table.add_rows([["Name", "Age", "Nickname"],
    ["Mr\nXavier\nHuon", 32, "Xav'"],
    ["Mr\nBaptiste\nClement", 1, "Baby"],
    ["Mme\nLouise\nBourgeau", 28, "Lou\n\nLoue"]])

for i in [4,14]:
    table.set_deco(i)
    print(table.draw())
    print()

$ python test.py
texttable version: 1.6.3

  Name     Age   Nickname
Mr                       
Xavier      32           
Huon               Xav'  
-------------------------
Mr                       
Baptiste     1           
Clement            Baby  
-------------------------
Mme                Lou   
Louise      28           
Bourgeau           Loue  

  Name   | Age | Nickname
=========+=====+=========
Mr       |     |         
Xavier   |  32 |         
Huon     |     |   Xav'  
-------------------------
Mr       |     |         
Baptiste |   1 |         
Clement  |     |   Baby  
-------------------------
Mme      |     |   Lou   
Louise   |  28 |         
Bourgeau |     |   Loue  
```

**Comment by paul-shuvo** [comment]
Could you try the following:
```python
for i in [13,14]:
    table.set_deco(i)
    print(table.draw())
    print()
```
I think it probably retains some configuration values from the previous `_deco` and doesn't reset it.

**Comment by foutaise** [comment]
Hi,

Ok, that was a cache issue. Fixed in release 1.6.4. (git commit https://github.com/foutaise/texttable/commit/98657cc2a25b6ba2abe6ee86a50ad3ef90f403b6).
Thanks for your feedback.

Best regards.
